test/WebJobs.Script.Tests.E2E/GeneralEndToEndTests.cs

```diff
@@ -136,14 +138,63 @@ public GeneralEndToEndTests(FunctionAppFixture fixture)
                     {
                         break;
                     }
-                    
+
                     await Task.Delay(3000);
                 }
 
                 _fixture.Assert.True(resultToken != null);
             }
         }
 
+        [Fact]
+        [TestTrace]
+        public async Task ServiceBus_Node_DoesNotExhaustConnections()
+        {
+            var connectionString = Environment.GetEnvironmentVariable("AzureWebJobsServiceBus");
+            NamespaceManager manager = NamespaceManager.CreateFromConnectionString(connectionString);
+
+            // Start with an empty queue
+            await manager.DeleteQueueAsync("node");
+
+            int i = 0, j = 0, lastConnectionCount = 0, lastConnectionLimit = 0;
+
+            using (var client = CreateClient())
+            {
+                // max connections in dynamic is currently 300
+                for (i = 1; i <= 25; i++)
+                {
+                    for (j = 1; j <= 25; j++)
+                    {
+                        List<Task<HttpResponseMessage>> requestTasks = new List<Task<HttpResponseMessage>>();
+                        requestTasks.Add(client.GetAsync($"api/ServiceBusNode?code={_fixture.FunctionDefaultKey}"));
+
+                        await Task.WhenAll(requestTasks);
```

Comment:
Did you want this `WhenAll` to be outside of the scope of this loop? We'll only have one task here at this point.